src/deploy/functions/validate.ts

```diff
@@ -126,6 +127,23 @@ function assertFunctionsSourcePresent(data: any, sourceDir: string, projectDir:
   }
 }
 
+/**
+ * Asserts that package.json does not include any scripts with invalid names
+ * @param data Object representing package.json file.
+ * @throws { FirebaseError } package.json must not contain scripts with protected names
+ */
+function checkForProtectedNpmScriptNames(data: any): void {
+  const protectedScriptNames = ["prepare"];
+  if (Object.prototype.hasOwnProperty.call(data, "scripts")) {
```

Comment:
While this is technically safer, I prefer this as the simpler to read:
```suggestion
  if (data?.scripts)) {
```

Reply:
Furthermore, if we do this, we can simplify this whole chunk even further, to something like:
```suggestion
    for (const scriptName of protectedScriptNames) {
      if (data?.scripts?.[scriptName])) {
        throw new FirebaseError(`package.json contains script with protected name ${scriptName}`);
      }
    }
```

Reply:
@joehan Thanks for the suggestions. I agree with you that stylistically you approach is better. It frames the fact that _data_ can be null/undefined better than my approach. 

The only downside I see is the fact that we are testing for the existence of _data_ and _data.scripts_ with every iteration of the loop. Right now this makes no difference since we only have one "protected word" (I'm still unsure about the term haha). If we do find other script names that conflict with the deploy we would be needlessly running checks for all of them even if _data_ or _data.scripts_ is not available.

I guess it boils down to project preference...